jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudformation/CloudFormationTemplate.kt

```diff
@@ -114,4 +118,28 @@ fun List<Parameter>.mergeRemoteParameters(remoteParameters: List<software.amazon
             mutableParameter.setDefaultValue(it.parameterValue())
         }
         mutableParameter
-    }.toList()
\ No newline at end of file
+    }.toList()
+
+/**
+ * Validate whether the specified template is supported for deployment to AWS.
+ *
+ * @return null if it is supported, or an error message otherwise.
+ */
+fun Project.canDeployOrError(path: String): String? {
```

Comment:
Opinion: `canX` functions at first glance would be expected to return Boolean. You've documented the return (awesome!)  but I wonder if a function that returns error text else null on success is better named `validateX`